src/extractor/guidance/intersection_handler.cpp

```diff
@@ -383,136 +393,140 @@ std::size_t IntersectionHandler::findObviousTurn(const EdgeID via_edge,
+    // straight and is same or higher priority class
+    const auto sameOrHigher = [&](const out_way &way) {
+        return way.same_or_higher_priority == true;
+    };
+    auto best_candidate_it = std::find_if(begin(out_ways), End, sameOrHigher);
+
+    // if no candidate is found with same or higher priority, use first non-low priority class
+    const auto notLowPriority = [&](const out_way &way) {
+        return !node_based_graph.GetEdgeData(way.road->eid)
+                    .road_classification.IsLowPriorityRoadClass();
+    };
+    if (best_candidate_it == End)
+        best_candidate_it = std::find_if(begin(out_ways), End, notLowPriority);
+
+    if (best_candidate_it != End)
+    {
+        best = best_candidate_it->index;
+        best_deviation = best_candidate_it->deviation_from_straight;
+    }
 
-        if ((!other_is_better_choice_by_priority && deviation < best_deviation) ||
-            is_better_choice_by_priority)
+    // double check if the sorted lowest deviation from straight would still be better choice
+    if (static_cast<uint32_t>(out_ways[0].index) != best)
+    {
+        const out_way first = out_ways[0];
+        const auto deviation_diff =
+            (best_deviation - first.deviation_from_straight) > FUZZY_ANGLE_DIFFERENCE;
+        const auto not_ramp_class = !node_based_graph.GetEdgeData(intersection[first.index].eid)
+                                         .road_classification.IsRampClass();
+        const auto not_link_class = !node_based_graph.GetEdgeData(intersection[first.index].eid)
+                                         .road_classification.IsLinkClass();
+        if (deviation_diff && notLowPriority(first) && not_ramp_class && not_link_class)
         {
-            best_deviation = deviation;
-            best = i;
+            best = first.index;
+            best_deviation = first.deviation_from_straight;
         }
     }
 
-    // We don't consider empty names a valid continue feature. This distinguishes between missing
-    // names and actual continuing roads.
-    if (in_data.name_id == EMPTY_NAMEID)
-        best_continue = 0;
-
+    // No non-low priority roads? Declare no obvious turn
     if (best == 0)
         return 0;
 
-    const std::pair<std::int64_t, std::int64_t> num_continue_names = [&]() {
-        std::int64_t count = 0, count_valid = 0;
-        if (in_data.name_id != EMPTY_NAMEID)
-        {
-            for (std::size_t i = 1; i < intersection.size(); ++i)
-            {
-                const auto &road = intersection[i];
-                const auto &road_data = node_based_graph.GetEdgeData(road.eid);
-
-                const auto same_name = !util::guidance::requiresNameAnnounced(
-                    in_data.name_id, road_data.name_id, name_table, street_name_suffix_table);
+    // best_continue is the first sorted way that shares the same name as the in way
+    const auto sameName = [&](const out_way &lhs) { return lhs.same_name_id == true; };
```

Comment:
from all these checks, I get the impression that the only sorting you require above is actually the `angularDifference`?